packages/contracts-bedrock/src/L2/L1BlockHolocene.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { L1BlockInterop, ConfigType } from "src/L2/L1BlockInterop.sol";
+import { StaticConfig } from "src/libraries/StaticConfig.sol";
+import "src/libraries/L1BlockErrors.sol";
+
+/// @custom:proxied
+/// @custom:predeploy 0x4200000000000000000000000000000000000015
+/// @title L1BlockHolocene
+/// @notice Interop extenstions of L1Block.
+contract L1BlockHolocene is L1BlockInterop {
+    /// @notice Event emitted when a new batcher hash is set.
+    event BatcherHashSet(bytes32 indexed batcherHash);
+
+    /// @notice Event emitted when a new ecotone gas config is set.
+    event GasConfigEcotoneSet(uint32 indexed blobBaseFeeScalar, uint32 indexed baseFeeScalar);
+
+    /// @notice Event emitted when a new gas limit is set.
+    event GasLimitSet(uint64 indexed gasLimit);
+
+    /// @notice The gas limit of L2 blocks in the same epoch.
+    uint64 public gasLimit;
+
+    /// @notice Sets static configuration options for the L2 system. Can only be called by the special
+    ///         depositor account.
+    /// @param _type  The type of configuration to set.
+    /// @param _value The encoded value with which to set the configuration.
+    function setConfig(ConfigType _type, bytes calldata _value) external override {
+        if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
+
+        if (_type == ConfigType.SET_GAS_PAYING_TOKEN) {
+            _setGasPayingToken(_value);
+        } else if (_type == ConfigType.ADD_DEPENDENCY) {
+            _addDependency(_value);
+        } else if (_type == ConfigType.REMOVE_DEPENDENCY) {
+            _removeDependency(_value);
```

Comment:
not a solidity expert can/should you call "super.setConfig(_type, _value)" instead of duplicating this code here?

Reply:
Note: `L1BlockHolocene` will be shipped before `L1BlockInterop`, and `L1BlockInterop.setConfig` is `external` instead of `public`, so not callable from inherited functions.